src/common/QskBoxBorderColors.cpp

```diff
@@ -61,105 +64,104 @@ QskBoxBorderColors::~QskBoxBorderColors()
 
 bool QskBoxBorderColors::operator==( const QskBoxBorderColors& other ) const
 {
-    return ( m_colors[ 0 ] == other.m_colors[ 0 ] ) &&
-           ( m_colors[ 1 ] == other.m_colors[ 1 ] ) &&
-           ( m_colors[ 2 ] == other.m_colors[ 2 ] ) &&
-           ( m_colors[ 3 ] == other.m_colors[ 3 ] );
+    return ( m_gradients[ 0 ] == other.m_gradients[ 0 ] ) &&
+           ( m_gradients[ 1 ] == other.m_gradients[ 1 ] ) &&
+           ( m_gradients[ 2 ] == other.m_gradients[ 2 ] ) &&
+           ( m_gradients[ 3 ] == other.m_gradients[ 3 ] );
 }
 
 void QskBoxBorderColors::setAlpha( int alpha )
 {
     for ( int i = 0; i < 4; i++ )
     {
-        if ( m_colors[ i ].isValid() && m_colors[ i ].alpha() )
-            m_colors[ i ].setAlpha( alpha );
+        if ( m_gradients[ i ].isValid() )
+            m_gradients[ i ].setAlpha( alpha );
     }
 }
 
-void QskBoxBorderColors::setColors( const QColor& color )
+void QskBoxBorderColors::setGradients( const QskGradient& gradient )
 {
-    qskSetColors( color, m_colors );
+    qskSetGradients( gradient, m_gradients );
 }
 
-void QskBoxBorderColors::setColors(
-    const QColor& left, const QColor& top,
-    const QColor& right, const QColor& bottom )
+void QskBoxBorderColors::setGradients( const QskGradient& left, const QskGradient& top,
+    const QskGradient& right, const QskGradient& bottom )
 {
-    qskSetColors( left, top, right, bottom, m_colors );
+    qskSetGradients( left, top, right, bottom, m_gradients );
 }
 
-void QskBoxBorderColors::setColor(
-    Qsk::Position position, const QColor& color )
+void QskBoxBorderColors::setGradient( Qsk::Position position, const QskGradient& gradient )
 {
-    m_colors[ position ] = color.toRgb();
+    m_gradients[ position ] = gradient;
 }
 
-void QskBoxBorderColors::setColorsAt( Qt::Edges edges, const QColor& color )
+void QskBoxBorderColors::setGradientAt( Qt::Edges edges, const QskGradient& gradient )
 {
-    const QColor c = color.toRgb();
-
     if ( edges & Qt::TopEdge )
-        m_colors[ Qsk::Top ] = c;
+        m_gradients[ Qsk::Top ] = gradient;
 
     if ( edges & Qt::LeftEdge )
-        m_colors[ Qsk::Left ] = c;
+        m_gradients[ Qsk::Left ] = gradient;
 
     if ( edges & Qt::RightEdge )
-        m_colors[ Qsk::Right ] = c;
+        m_gradients[ Qsk::Right ] = gradient;
 
     if ( edges & Qt::BottomEdge )
-        m_colors[ Qsk::Bottom ] = c;
+        m_gradients[ Qsk::Bottom ] = gradient;
 }
 
-QColor QskBoxBorderColors::colorAt( Qt::Edge edge ) const
+QskGradient QskBoxBorderColors::gradientAt( Qt::Edge edge ) const
```

Comment:
const QskGradient& - a copy is not too expensive, as the stops are a QVector, but as this class is used in the box renderer I would avoid it. For the situation, where no edge has been selected I would write:

    static QskGradient noGradient;
    return noGradient;

Reply:
You meant that the method should return a const ref, right? If yes, then the code is fine now.